core/src/main/java/discord4j/core/object/util/PermissionSet.java

```diff
@@ -87,54 +98,126 @@ private PermissionSet(final long rawValue) {
     }
 
     /**
-     * Performs a logical AND of of this permission set with the other permission set.
+     * Performs a logical <b>AND</b> of this permission set with the other permission set.
+     * <p>
+     * The resultant set is the <b>intersection</b> of this set and the other set. A permission is contained iff it was
+     * contained in both this set and the other set.
+     * <pre>
+     * {@code
+     * PermissionSet set0 = PermissionSet.of(KICK_MEMBERS, BAN_MEMBERS);
+     * PermissionSet set1 = PermissionSet.of(KICK_MEMBERS);
+     * set0.and(set1) = PermissionSet.of(KICK_MEMBERS);
+     * }
+     * </pre>
      *
      * @param other The other permission set.
-     * @return A new permission set of this set AND the other set.
+     * @return The intersection of this set with the other set.
      */
     public PermissionSet and(PermissionSet other) {
         return PermissionSet.of(this.rawValue & other.rawValue);
     }
 
     /**
-     * Performs a logical OR of this permission set with the other permission set.
+     * Performs a logical <b>OR</b> of this permission set with the other permission set.
+     * <p>
+     * The resultant set is the <b>union</b> of this set and the other set. A permission is contained iff it was
+     * contained in either this set or the other set.
+     * <pre>
+     * {@code
+     * PermissionSet set0 = PermissionSet.of(KICK_MEMBERS);
+     * PermissionSet set1 = PermissionSet.of(BAN_MEMBERS);
+     * set0.and(set1) = PermissionSet.of(KICK_MEMBERS, BAN_MEMBERS);
+     * }
+     * </pre>
      *
      * @param other The other permission set.
-     * @return A new permission set of this set OR the other set.
+     * @return The union of this set with the other set.
      */
     public PermissionSet or(PermissionSet other) {
         return PermissionSet.of(this.rawValue | other.rawValue);
     }
 
     /**
-     * Performs a logical NOT of this permission set.
+     * Performs a logical <b>XOR</b> of this permission set with the other permission set.
+     * <p>
+     * The resultant set is the <b>symmetric difference</b> of this set and the other set. A permission is contained iff
+     * it was contained in <b>only</b> this set or contained in <b>only</b> the other set.
+     * <pre>
+     * {@code
+     * PermissionSet set0 = PermissionSet.of(KICK_MEMBERS, BAN_MEMBERS, ATTACH_FILES);
+     * PermissionSet set1 = PermissionSet.of(ATTACH_FILES, CONNECT);
+     * set0.xor(set1) = PermissionSet.of(Permission.KICK_MEMBERS, Permission.BAN_MEMBERS, Permission.CONNECT)
+     * }
+     * </pre>
      *
-     * @return A new permission set representing this set's complement.
+     * @param other The other permission set.
+     * @return The symmetric difference of this set with the other set.
      */
-    public PermissionSet not() {
-        return PermissionSet.of(~this.rawValue);
+    public PermissionSet xor(PermissionSet other) {
```

Comment:
Is XOR a valuable function at all? I can't think of a single case where it would be useful for manipulating permissions.

Reply:
In general, XOR is useful for toggling bits in a bit field. A use with permissions doesn't immediately come to mind, but I don't see a problem in providing basic operations, especially when it would be relatively difficult for a user to implement.